src/interpreter/executor/interpret.js

```diff
@@ -0,0 +1,332 @@
+    ACCEPTING_VAR : "accepting variable",
+    ASSIGNING_VAR : "assigning variable"
+}
+
+function Variable() {
+    this.name = "";
+    this.value = "";
+}
+
+// Our master state that we will output to visualizer
+var masterRep = [];
+
+// Single line representation we will push to master Rep
+var currentLineRep = [];
+
+// Current interpreter state for program
+var currentState = stateEnum.DEFAULT;
+
+// Current step in the user input code
+var currentStep = 0;
+
+// Current object we are adding data to
+var currentDataHold = null;
+
+//  Flag to tell code editor to be writable or not
+var isRunning = false;
+
+// Global var to bump out of interpreting if in a block comment
+var isBlockComment = false;
+
+//  List to hold user code, use a list so we can index on stepCount
+var userCode = []
+
+// Setup for when we initially check syntax
+function setup(inputCode) {
+    
+    //Lock  the code editor
+    isRunning = true;
+    
+    //Check validity of inputCode first
+    try {
+        new Function(inputCode);
+    } catch(err) {
+        exitInterpret();
+        return false;
+    }
+    
+    userCode = inputCode.split(/[\r\n]+/)
+    return true;
+}
+
+// Cleanup function to be called whenever exiting interpreter
+function exitInterpret() {
+    
+    masterRep = [];
+    currentLineRep = [];
+    currentState = stateEnum.DEFAULT;
+    currentStep = 0;
+    isRunning = false;
+    userCode = [];
+    isBlockComment = false;
+    currentDataHold = null;
+}
+
+function interpretLine(lineNumber) {
+
+    //Handling multi statements on one line
+    var statements = userCode[lineNumber].split(';');
+    
+    for (var i = 0; i < statements.length; i++) {
+        var statement = statements[i];
+        if (statement.trim() == "") {
+            continue;
+        }
+
+        //Fresh buffer for each statement
+        var buffer = "";
+
+        // Iterate each character in the current user code line
+        for (var j = 0; j < statement.length; j++) {
+            if (statement[j] == ' ') {
+                continue;
+            }
+    
+            buffer += statement[j]
+            
+            //Handling comments here
+            if (isBlockComment && buffer == '*') {
+                continue;
+            }
+            if (handleComments(buffer) || isBlockComment) {
+                buffer = "";
+                break;
+            }
+
+            //Skip on cases not yet covered yet
+            if (isNotCovered(buffer)) {
+                buffer = "";
+                break;
+            }
+
+            if (currentState == stateEnum.DEFAULT) {
+                buffer = findBufferState(buffer);
+            } else {
+                buffer = evalState(buffer);
+            }        
+        }
+        resolveState(buffer);
+    }
+
+    masterRep.push([lineNumber, currentLineRep]);
+    postInterpretCleanup();
+}
+
+
+// Beginning basic comment support
+function handleComments(buffer) {
+
+    if (buffer == "//") {
```

Comment:
I think this is a case to use strict identity equality to compare the strings, jic

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Comparison_Operators#Identity

Reply:
Didn't know about this. Thanks!